src/components/embedWrapper.tsx

```diff
@@ -0,0 +1,243 @@
+// ----- Imports ----- //
+
+import { EmbedTracksType } from '@guardian/content-api-models/v1/embedTracksType';
+import type { Result } from '@guardian/types';
+import { err, fromNullable, map, none, ok, withDefault } from '@guardian/types';
+import { andThen } from '@guardian/types/dist/result';
+import { ClickToView } from 'components/ClickToView';
+import EmbedComponent from 'components/embed';
+import { EmbedKind } from 'embed';
+import type { Embed, Generic, Spotify, YouTube } from 'embed';
+import { resultFromNullable, resultMap2, resultMap3 } from 'lib';
+import { createElement as h } from 'react';
+import type { FC, ReactElement } from 'react';
+
+// ----- Component ----- //
+
+interface Props {
+	embed: Embed;
+}
+
+const embedToDivProps = (embed: Embed): Record<string, string> => {
+	switch (embed.kind) {
+		case EmbedKind.Spotify:
+			return {
+				kind: EmbedKind.Spotify,
+				src: embed.src,
+				width: embed.width.toString(),
+				height: embed.height.toString(),
+				...withDefault<Record<string, string>>({})(
+					map<string, Record<string, string>>((source) => {
+						return { source };
+					})(embed.source),
+				),
```

Comment:
An alternative way to write this. If you find it less clear feel free to ignore 🙂:

```suggestion
				...pipe2(
					embed.source,
					map((source) => ({ source })),
					withDefault<Record<string, string>>({}),
				),
```

Since you're using this pattern a few times here it may be worth pulling out into a function?

Reply:
Nice :)